File: CH9/StartWithJEE8/StartWithJEE8-ejb/src/main/java/com/daniel/delivery/abstraction/entity/FoodService.java
package com.daniel.delivery.abstraction.entity;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

public class FoodService implements Serializable {

    private final String id;
    private final String email;
    private final String name;
    private final String address;
    private final String imageUrl;
    private final String foodType;
    private final int deliveryFee;
    private boolean active;
    private final User user;
    private final List<FoodProduct> foodProductList;

    public FoodService(String id, String email, String name, String address, String imageUrl, String foodType, int deliveryFee, boolean active, User user, List<FoodProduct> foodProductList) {
        this.id = id;
        this.email = email;
        this.name = name;
        this.address = address;
        this.imageUrl = imageUrl;
        this.foodType = foodType;
        this.active = active;
        this.foodProductList = foodProductList;
        this.deliveryFee = deliveryFee;
        this.user = user;
    }

    public String getId() {
        return id;
    }

    public void deActivate(){
        this.active = false;
    }
    
    public String getEmail() {
        return email;
    }

    public String getName() {
        return name;
    }

    public String getAddress() {
        return address;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public String getFoodType() {
        return foodType;
    }

    public int getDeliveryFee() {
        return deliveryFee;
    }

    public boolean getActive() {
        return active;
    }

    public User getUser() {
        return user;
    }

    public List<FoodProduct> getFoodProductList() {
        return foodProductList;
    }

    @Override
    public int hashCode() {
        int hash = 3;
        hash = 59 * hash + Objects.hashCode(this.id);
        hash = 59 * hash + Objects.hashCode(this.email);
        hash = 59 * hash + Objects.hashCode(this.name);
        hash = 59 * hash + Objects.hashCode(this.address);
        hash = 59 * hash + Objects.hashCode(this.imageUrl);
        hash = 59 * hash + Objects.hashCode(this.foodType);
        hash = 59 * hash + this.deliveryFee;
        hash = 59 * hash + (this.active ? 1 : 0);
        hash = 59 * hash + Objects.hashCode(this.user);
        hash = 59 * hash + Objects.hashCode(this.foodProductList);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final FoodService other = (FoodService) obj;
        if (this.deliveryFee != other.deliveryFee) {
            return false;
        }
        if (this.active != other.active) {
            return false;
        }
        if (!Objects.equals(this.id, other.id)) {
            return false;
        }
        if (!Objects.equals(this.email, other.email)) {
            return false;
        }
        if (!Objects.equals(this.name, other.name)) {
            return false;
        }
        if (!Objects.equals(this.address, other.address)) {
            return false;
        }
        if (!Objects.equals(this.imageUrl, other.imageUrl)) {
            return false;
        }
        if (!Objects.equals(this.foodType, other.foodType)) {
            return false;
        }
        if (!Objects.equals(this.user, other.user)) {
            return false;
        }
        if (!Objects.equals(this.foodProductList, other.foodProductList)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "FoodService{" + "id=" + id + ", email=" + email + ", name=" + name + ", address=" + address + ", imageUrl=" + imageUrl + ", foodType=" + foodType + ", deliveryFee=" + deliveryFee + ", active=" + active + ", user=" + user + ", foodProductList=" + foodProductList + '}';
    }  
    
}

File: CH8/startwithvuejs/src/entities/FoodService.ts
import { User } from './User'

export class FoodService {
    email: string = ''
    name: string = ''
    address: string = ''
    foodType: string = ''
    deliveryFee: number = 0.0
    image: any | null = null
    imageUrl: string = ''
    user: User = User.emptyUser()
    active: boolean = false

    static emptyFoodService () {
      let foodService:FoodService = new FoodService()

      return foodService
    }

    static newFoodService (email: string, name: string, address: string, foodType: string, deliveryFee: number, image: any, imageUrl: string, user: User, active: boolean) {
      let foodService:FoodService = new FoodService()

      foodService.email = email
      foodService.name = name
      foodService.address = address
      foodService.foodType = foodType
      foodService.deliveryFee = deliveryFee
      foodService.image = image
      foodService.imageUrl = imageUrl
      foodService.user = user
      foodService.active = active

      return foodService
    }

    copyFoodService (foodServiceToCopy: FoodService) {
      this.email = foodServiceToCopy.email
      this.name = foodServiceToCopy.name
      this.address = foodServiceToCopy.address
      this.foodType = foodServiceToCopy.foodType
      this.deliveryFee = foodServiceToCopy.deliveryFee
      this.image = foodServiceToCopy.image
      this.imageUrl = foodServiceToCopy.imageUrl
      this.user = foodServiceToCopy.user
      this.active = foodServiceToCopy.active
    }
}
